src/Facebook/FacebookBatchRequest.php

```diff
@@ -189,8 +191,19 @@ public function convertRequestsToJson()
     {
         $requests = [];
         foreach ($this->requests as $request) {
-            $attachedFiles = isset($request['attached_files']) ? $request['attached_files'] : null;
-            $requests[] = $this->requestEntityToBatchArray($request['request'], $request['name'], $attachedFiles);
+            $options = [];
+
+            if (!empty($request['name'])) {
```

Comment:
@teldosas I'm talking about the `$key` variable in `add()`, the case when you give an array of requests.
In `->add($myRequest)`, `$name` will be null
In `->add($ManyRequests)`, `$name` for eachrequest will be its key in the array, an integer, which is not allowed by our doc, that's why I suggest you to `->add($request, strval($key))` in the loop :)